ui/safe-offset.js

```diff
@@ -0,0 +1,25 @@
+( function( factory ) {
+	if ( typeof define === "function" && define.amd ) {
+
+		// AMD. Register as an anonymous module.
+		define( [ "jquery", "./version" ], factory );
+	} else {
+
+		// Browser globals
+		factory( jQuery );
+	}
+} ( function( $ ) {
+return $.ui.__safeOffset__ = function( element ) {
+
+	// Simulate a jQuery short-circuiting when there are no client rects reported
+	// which usually means a disconnected node. This check in jQuery is meant just
+	// for IE but UI depends on it.
```

Comment:
Just for my understanding: this check is implemented in jQuery for IE only, bit UI needs it in every browser, right?

Reply:
Not exactly. jQuery has a check meant for IE but it's run by all browsers so by pure coincidence it makes UI not hit errors here. This is about these lines:
https://github.com/jquery/jquery/blob/3.5.1/src/offset.js#L98-L104

Reply:
How about:
```suggestion
	// for IE but is applied in all browsers & UI depends on it.
```
?

Reply:
I still don't understand why we need this in jQueryUI when its already in jQuery... sorry, my mind is a little slow today ;)

Reply:
See https://jquery.com/upgrade-guide/3.0/#breaking-change-invalid-input-to-the-offset-method

jQuery 3.0.0 & newer require DOM elements with the `getBoundingClientRect` method. It's just that jQuery needs to run a check for `getClientRects()` due to the comment above the lines I linked, to workaround an IE issue. We shouldn't rely on this check for other purposes.

Also, jQuery Migrate warns if you use `offset()` in such a way.

That said, now I see the discussion in https://github.com/jquery/jquery/issues/2310 so maybe I need to re-read it & make sure Migrate is not warning too much.

Reply:
Ahhh, got it. Thanks for the explanation and your patience :-)

Reply:
@fnagel Thank you for raising these concerns! This change will most likely not be needed as we'll update Migrate instead.